### DISCUSSION TITLE
Converting SQL Agent Jobs over from SQLCMD to TSQL

### DISCUSSION THREAD

**Discussion by erichanson76** [olahallengren/sql-server-maintenance-solution#208] [Questions & Answers]
Looking to see if there's a solution anyone has already put together for upgrading from an older version of the maintenance solution to the newer version - to also update the jobs from sqlcmd to TSQL (where the OS is Windows 2012 or higher).  The upgrade certainly upgrades the procs and the jobs still work.  We can write up something if there isn't something available. We'd want to maintain what is there (ie if someone has included / excluded / etc) but just take out the unnecessary characters / convert the jobs over as needed.

**Comment by olahallengren** [comment]
I don't have any scripts to do this.

I should also add that CmdExec job steps with sqlcmd still works well.

The reason that I started to use T-SQL job steps in SQL Server 2012 is that the original issue with T-SQL job steps was fixed. In addition you can use multiple lines and they are easier to read.
